lib/helpers.js

```diff
@@ -21,20 +21,16 @@ const secureHosts = require('./secureHosts.json');
  * @return {string} processed URL
  */
 const upgradeToHttps = (urlString) => {
-  // Add a scheme to scheme-less URLs
-  const isRelative = Boolean(urlString.match(/^\/\//));
+  // For URLs that are protocol-relative or missing it completely
+  const missingScheme = urlString.indexOf(":") === -1;
+  const bareHostname = Boolean(urlString.match(/^[a-zA-Z0-9]$/));
 
-  if (isRelative) {
-    urlString = `https:${urlString}`;
+  if (missingScheme || bareHostname) {
+    urlString = `http:${urlString}`;
```

Comment:
This changes the previous behavior though. Before we defaulted to `https`.

Reply:
Yeah, that's an oddity I'd like to sort out:
In https://github.com/mozilla/srihash.org/blob/master/test/helpers.js#L30-L31, we expect `example.com/script.js` to be completed to HTTP, but below in https://github.com/mozilla/srihash.org/blob/master/test/helpers.js#L37-L38 we autofill the protocol-relative `//example.com/script.js` to HTTPS.

It's not in secureHosts.json, so why do we upgrade in the latter case?

Reply:
Hmm, yeah that doesn't seem right indeed. On a side note, since the `input` has `type="url"`, the user doesn't seem to be able to enter a protocol-relative file. At least with FF 66 that I use.

Reply:
This changes was done in https://github.com/mozilla/srihash.org/pull/156

So, if we trust the browsers' checks here, I think we can just drop the whole relative URLs logic (protocol and root relative).

Reply:
We shouldn't trust the browser completely. We should try to parse with the URL constructor and return an error to the user interface if it throws

Reply:
Then why did you merge #233 :P

You can add back any stuff here after rebasing this.

BTW you should open branches upstream so that it's easier to collaborate. I will start doing it too as long as we sort #234

Reply:
It seems I love rebasing after lunch.